### DISCUSSION TITLE
Improve Drag-and-Drop UX

### DISCUSSION THREAD

**Discussion by djmtype** [directus/directus#4496] [Feature Requests]
Please consider improving drag-n-drop. It's difficult to tell were modules are landing within the admin area.

**Comment by benhaynes** [comment]
We'll look into improving this, but any additional insight into how it could be improved would be appreciated!

**Comment by djmtype** [comment]
In the most recent stable versions of Firefox and Chrome, after dropping the field, more times than not, it's automatically triggering the field's options. In other worlds, it's opening the side modal. Safari doesn't suffer from this issue. 
https://cloudup.com/cBdv3W-mDkN

Tested with Directus 9 rc44 and rc49 using incognito mode on all browsers so my plugins don't interfere.

I also wanted to address the visual aspects of the drag-n-drop component used throughout Directus.

Currently, the object being grabbed is more transparent than the dropzone. If you look at this from a 3D perspective, the foreground object (the field being grabbed) should be more opaque and the background (the dropzone) more muted. In Directus, it's the opposite. IMO, it's creating a visual clash, and not easily distinguishable between foreground and background objects.

[vuedraggable](https://sortablejs.github.io/Vue.Draggable/#/simple) is a library that pulls off a nice representation of drag-n-drop. Notice when you pick up an item, the item being grabbed more opaque. It uses a drop shadow to give it the illusion of depth. The entire drop zone is represented in a different, but muted color to give a better sense of separation. The dropzone label has also been muted.

The same library demonstrates another variation which relies on animation for visual feedback, while muting the dropzone area. [Example with animation](https://sortablejs.github.io/Vue.Draggable/#/transition-example-2)

[This Slicksort example](https://jexordexan.github.io/vue-slicksort/?selectedKind=Vertical%20sorting&selectedStory=Simple%20list&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) relies on animation for visual cues. In this specific example, its dropzone color is the same as its background color. What I like is that it doesn't use a muted label of its former self which reduces visual clutter. 
Here's the official Slicksort library: https://vue-slicksort.netlify.app/

I am not suggesting you replace your drag-n-drop component with one of these libraries. Just build upon what you already have and hopefully get feedback from other users.

Thanks for creating this awesome CMS!

**Reply by rijkvanzanten** [reply]
> vuedraggable is a library that pulls off a nice representation of drag-n-drop. Notice when you pick up an item, the item being grabbed more opaque. It uses a drop shadow to give it the illusion of depth. The entire drop zone is represented in a different, but muted color to give a better sense of separation. The dropzone label has also been muted.

Funny that you mention this. We currently rely on `vuedraggable`, and the quirks you see are the quirks that come straight of that library 😬 Definitely gonna take a look at SlickSort though!

**Reply by djmtype** [reply]
Maybe it's just the CSS then?

**Reply by rijkvanzanten** [reply]
Yeah, those animations are cause we don't wrap things in a transition-group, but vuedraggable has generally turned out to be pretty finicky unfortunately

**Reply by djmtype** [reply]
Things become finicky too when you mixed rows of 2 items – kinda like rearranging your app icons' order on an iOS screen.

**Reply by rijkvanzanten** [reply]
It got especially tricky when working on a nested tree-view setup the other day, goody goody i need some time to recover from that 😅

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
